### PR TITLE
[MAMAEdu-75] Modify API endpoint to enroll by email

### PR DESCRIPTION
**Pull request by dyudyunov**
**Changes:**
- endpoint modified to enroll by email
- test for enrolling by email
- fix for enrolling views tests

**Youtrack:**
https://youtrack.raccoongang.com/issue/MAMAEdu-75

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: common/djangoapps/enrollment/tests/test_views.py [hunk 1]
```diff
@@ -47,14 +47,15 @@ def assert_enrollment_status(
...
+            email=None,
             expected_status=status.HTTP_200_OK,
             email_opt_in=None,
             as_server=False,
             mode=CourseMode.DEFAULT_MODE_SLUG,
             is_active=None,
             enrollment_attributes=None,
             min_mongo_calls=0,
-            max_mongo_calls=0,
+            max_mongo_calls=1,  # see student.models.py::1235
```
Comment (OPersian) [inline common/djangoapps/enrollment/tests/test_views.py:58]: What if the row number changes

Comment (OPersian) [reply common/djangoapps/enrollment/tests/test_views.py:58]: maybe more informative reference would be in place

File: common/djangoapps/enrollment/views.py [hunk 2]
```diff
@@ -555,12 +556,16 @@ def post(self, request):
...
+                user = User.objects.get(username=username)
+            else:
+                user = User.objects.get(email=email)
+                username = user.username
         except ObjectDoesNotExist:
             return Response(
                 status=status.HTTP_406_NOT_ACCEPTABLE,
                 data={
-                    'message': u'The user {} does not exist.'.format(username)
+                    'message': u'The user {} does not exist.'.format(email if email else username)
```
Comment (OPersian) [inline common/djangoapps/enrollment/views.py:568]: how about `email or username`

File: common/djangoapps/enrollment/tests/test_views.py (hunk 1 above)
Comment (dyudyunov) [reply common/djangoapps/enrollment/tests/test_views.py:58]: deleted in https://github.com/raccoongang/edx-platform/pull/2069

File: common/djangoapps/enrollment/views.py (hunk 2 above)
Comment (dyudyunov) [reply common/djangoapps/enrollment/views.py:568]: fixed in https://github.com/raccoongang/edx-platform/pull/2069

**Pull request merged**